### PR TITLE
Snippets, syntax coloring and resource files template

### PR DESCRIPTION
**Pull request by acauan**

### CODE REVIEW

**Comment by kaestlyn** [comment]
I'll review these changes this afternoon

File: snippets/cm.json [hunk 1]
```diff
@@ -47,10 +47,107 @@
...
+    },
+    "Switch statement": {
+        "prefix": "switch",
+        "body": [
+            "switch ($1) {",
+            "\tcase $2: $3",
+            "\tdefault: $0",
+            "}"
+        ],
+        "description": "Private method declaration"
```
Comment (kaestlyn) [inline snippets/cm.json]: Description is incorrect

File: snippets/cm.json [hunk 2]
```diff
@@ -47,10 +47,107 @@
...
+    },
+    "Pln": {
+        "prefix": "pln",
+        "body": [
+            "pln($0);"
+        ],
+        "description": "Private method declaration"
```
Comment (kaestlyn) [inline snippets/cm.json]: Description incorrect

File: snippets/cm.json [hunk 3]
```diff
@@ -47,10 +47,107 @@
...
+    },
+    "Auto constructor": {
+        "prefix": "cauto",
```
Comment (kaestlyn) [inline snippets/cm.json:110]: would it make sens to have this just be "ctor" like C# has it? Probably what most people are used to?

File: src/cmUtils.ts [hunk 4]
```diff
@@ -357,4 +358,45 @@ public class {Class} {
...
         });
     }
+
+    static createResourceTemplate(uri: vscode.Uri) {
+        var template =
+`package {Package};
+
+$ {
+    english "";
+}`
```
Comment (kaestlyn) [inline src/cmUtils.ts]: Add a semi-color at the end of the line for consistency.

File: syntaxes/CM.plist [hunk 5]
```diff
@@ -740,7 +740,7 @@
 			<array>
 				<dict>
 					<key>match</key>
-					<string>\b(true|false|null)\b</string>
+					<string>\b(true|false|null|var|function)\b</string>
```
Comment (kaestlyn) [inline syntaxes/CM.plist]: Can you add var and func under <key>keywords</key> patterns instead as a new node like:

``` xml
<dict>
    <key>match</key>
    <string>\b(var|function)\b</string>
    <key>name</key>
    <string>meta.token.keyword.java</string>
</dict>
```

So it's tagged more correctly by the syntax highlighter (coping how C# is done in vscode)

**Review by kaestlyn — CHANGES_REQUESTED** [review]
See my review comments.

File: snippets/cm.json (hunk 1 above)
Comment (acauan) [reply snippets/cm.json]: Fixed via f6c647e0

File: snippets/cm.json (hunk 2 above)
Comment (acauan) [reply snippets/cm.json]: Fixed via f6c647e0

File: snippets/cm.json (hunk 3 above)
Comment (acauan) [reply snippets/cm.json:110]: There are two different snippets. `ctor` creates an empty parameterless constructor and `cauto` creates an auto constructor, which automatically expects all public fields as parameters. We have both.

File: src/cmUtils.ts (hunk 4 above)
Comment (acauan) [reply src/cmUtils.ts]: Fixed via f6c647e0

File: syntaxes/CM.plist (hunk 5 above)
Comment (acauan) [reply syntaxes/CM.plist]: Fixed via f6c647e0

File: snippets/cm.json (hunk 3 above)
Comment (kaestlyn) [reply snippets/cm.json:110]: Ahh didn't notice that!

**Review by kaestlyn — APPROVED** [review]

**Comment by kaestlyn** [comment]
LGTM - next week I'll work on pushing this out as a new version.

**Pull request merged**